integration-tests/suites/gperftools.go

```diff
@@ -0,0 +1,78 @@
+package suites
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+type GperftoolsTestSuite struct {
+	IntegrationTestSuiteBase
+}
+
+// IsArchSupported: at the moment only x86_64 builds have gperftools
+func (s *GperftoolsTestSuite) IsArchSupported() (bool, string) {
+	u := unix.Utsname{}
+	unix.Uname(&u)
+
+	// Exclude null bytes from comparison
+	arch := string(bytes.Trim(u.Machine[:], "\x00"))
+	return arch == "x86_64", arch
```

Comment:
I wonder if we can make this a bit more general purpose for use elsewhere in the tests

```golang

// suites/common/utils.go
import "slices"

func ArchSupported(supported ...string) (bool, string) {
        u := unix.Utsname{}
        unix.Uname(&u)

        arch := string(bytes.Trim(u.Machine[:], "\x00"))
        return slices.Contains(supported, arch), arch
}
```